Otherwise got the following build error with uClibc 0.9.30:
  parser.c: In function 'uc_mgr_scan_master_configs':
  parser.c:1138: error: 'versionsort' undeclared (first use in this function)
  parser.c:1138: error: (Each undeclared identifier is reported only once
  parser.c:1138: error: for each function it appears in.)

(should be fine with more recent uClibcs)

As a consequence, alsa-utils are also downgraded and alsa-utils/alsa-utils-exp10-workaround.patch
removed.

Index: buildroot/package/multimedia/alsa-lib/alsa-lib.mk
===================================================================
--- buildroot.orig/package/multimedia/alsa-lib/alsa-lib.mk	2012-02-21 16:43:24.000000000 +0100
+++ buildroot/package/multimedia/alsa-lib/alsa-lib.mk	2012-02-21 16:46:03.000000000 +0100
@@ -3,7 +3,7 @@
 # alsa-lib
 #
 #############################################################
-ALSA_LIB_VERSION = 1.0.24.1
+ALSA_LIB_VERSION = 1.0.23
 ALSA_LIB_SOURCE = alsa-lib-$(ALSA_LIB_VERSION).tar.bz2
 ALSA_LIB_SITE = ftp://ftp.alsa-project.org/pub/lib
 ALSA_LIB_INSTALL_STAGING = YES
Index: buildroot/package/multimedia/alsa-utils/alsa-utils.mk
===================================================================
--- buildroot.orig/package/multimedia/alsa-utils/alsa-utils.mk	2012-02-21 16:45:59.000000000 +0100
+++ buildroot/package/multimedia/alsa-utils/alsa-utils.mk	2012-02-21 16:46:03.000000000 +0100
@@ -3,7 +3,7 @@
 # alsa-utils
 #
 #############################################################
-ALSA_UTILS_VERSION = 1.0.24.2
+ALSA_UTILS_VERSION = 1.0.23
 ALSA_UTILS_SOURCE = alsa-utils-$(ALSA_UTILS_VERSION).tar.bz2
 ALSA_UTILS_SITE = ftp://ftp.alsa-project.org/pub/utils
 ALSA_UTILS_INSTALL_STAGING = YES
Index: buildroot/package/multimedia/alsa-utils/alsa-utils-exp10-workaround.patch
===================================================================
--- buildroot.orig/package/multimedia/alsa-utils/alsa-utils-exp10-workaround.patch	2012-02-21 16:43:24.000000000 +0100
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1,33 +0,0 @@
-From da14c5ab2a6e030c5c56c979b8eb3ca071890876 Mon Sep 17 00:00:00 2001
-From: Peter Korsgaard <jacmet@sunsite.dk>
-Date: Mon, 14 Mar 2011 09:19:18 +0100
-Subject: [PATCH] alsamixer: fix build on uClibc
-
-exp10 is a glibc extension, which isn't supported on uClibc. Luckily,
-exp10() is trivial to compute based on exp(), so add a wrapper for
-the uClibc case.
-
-Signed-off-by: Peter Korsgaard <jacmet@sunsite.dk>
----
- alsamixer/volume_mapping.c |    5 +++++
- 1 files changed, 5 insertions(+), 0 deletions(-)
-
-diff --git a/alsamixer/volume_mapping.c b/alsamixer/volume_mapping.c
-index 9cacad8..1c0d7c4 100644
---- a/alsamixer/volume_mapping.c
-+++ b/alsamixer/volume_mapping.c
-@@ -37,6 +37,11 @@
- #include <stdbool.h>
- #include "volume_mapping.h"
- 
-+#ifdef __UCLIBC__
-+/* 10^x = 10^(log e^x) = (e^x)^log10 = e^(x * log 10) */
-+#define exp10(x) (exp((x) * log(10)))
-+#endif /* __UCLIBC__ */
-+
- #define MAX_LINEAR_DB_SCALE	24
- 
- static inline bool use_linear_dB_scale(long dBmin, long dBmax)
--- 
-1.7.2.3
-
